### DISCUSSION TITLE
How to add a custom api?

### DISCUSSION THREAD

**Discussion by ndphuong** [supabase/supabase#5963] [General]
Hi,
I want to add a custom API to Supabase.
Or can add more logic to existed API.
How can I do it?

Thanks

**Comment by liaujianjie** [comment]
From what I know, there are mainly 2 ways to write custom APIs to interact with Supabase:
1. [Supabase Functions](https://supabase.com/docs/guides/database/functions) (which uses Postgres functions under the hood) — I find that this is generally sufficient if you only need to interact with your data on Supabase.
2. Writing your own APIs and hosting them somewhere else. For example, on Vercel.

**Reply by ndphuong** [reply]
Hi @liaujianjie ,
About solution 2, can I integrate my own API with auth of Supabase?

**Reply by liaujianjie** [reply]
Can you provide more details? What does your integration do?

**Reply by Elvincth** [reply]
> Hi @liaujianjie ,
> 
> About solution 2, can I integrate my own API with auth of Supabase?
> 
> 

Same question, how do we make an auth middleware? Or how can we check the user auth?

**Comment by kk21** [comment]
Supabase functions-js coming?
https://github.com/supabase/supabase-js/releases/tag/v1.32.0
